### DISCUSSION TITLE
PHP version 7.4 in 9/2022 etc...

### DISCUSSION THREAD

**Discussion by iDad5** [docker-library/wordpress#750] [General]
I'm fairly new to docker, and I'm somehow undecided if I love or hat the experience. 

Setting up docker desktop following the instructions and installing the official WordPress built was really going smooth - aside from it taking me a while to figure out how to map the http-port not to collide with my local xampp...

Getting the the volumes to mount on windows was hell however for various reasons, first and foremost for a lack of good current documentation and tons of outdated info on the web...

My real issue however is, that the current package comes with a php version that is outdated - WordPress [recommendation ](https://wordpress.org/about/requirements/) for PHP 7.4 is a minimal and they linkt to 8.1 at the time of writing.

As in the provided docker-compose.yaml file there is no explicit mention of PHP I suppose it is somehow configured in the WordPress volume.  ? 

The WP recommendation is also HTTPS support, which is still not easy in a local dev-environment I know, but with no exposure of the apache config ist's even harder.

(I also would like to have PHP-MyAdmin and a way to configure PHP - is there a way to provide an alternative docker-composer.yaml to help with starting thing up in a more flexible environment?)

**Comment by wglambert** [comment]
You can specify `wordpress:6.0.2-php7.4` to get a WordPress version of 6.02 with PHP 7.4, we also have PHP 8.0 and 8.1 variants, see https://github.com/docker-library/docs/tree/master/wordpress#supported-tags-and-respective-dockerfile-links for the full list

For HTTPS support https://github.com/docker-library/wordpress/issues/46#issuecomment-304741999

Here's an example `docker-compose.yml` with Wordpress and Phpmyadmin https://github.com/docker-library/wordpress/issues/555 (switch `wordpress:cli` to a different variant)

**Reply by tianon** [reply]
See also https://make.wordpress.org/hosting/handbook/server-environment/#php, which notes that support for PHP 8.0 and 8.1 is still technically "beta" :sweat_smile:

**Reply by iDad5** [reply]
I feared as much....
